compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -342,6 +342,7 @@ void addDoubleTilingExpertPassPipeline(OpPassManager &passManager,
   // Add the sandbox single tiling expert to tile and vectorize.
   {
     LinalgSingleTilingExpertPassOptions options;
+    options.peel = true;
```

Comment:
If we add a new pipeline, we can control this option with by passing the value (true/false) from the caller. For the pipeline with peeling we use `true`.
I dont think peeling and padding go together right (at least for now)

Reply:
Yes. However, I think I've lost track of the current state of vectorization. Padding is only enabled for matmul but we added a few fixes to "always vectorized" (setting tile sizes to one) and other changes. Could you please give me a quick update? I understand that we should enable peeling for any non-matmul op, right?

> I dont think peeling and padding go together right (at least for now)

No, they wouldn't work together. I run this to get an overall performance picture an see how things evolve with the upcoming changes.

Reply:
Padding is only enabled for matmul. For other static cases, we set the tiling size to factors of dim sizes. If they are dynamic cases, the tiling sizes are set to one.

I think we can add a new pipeline for peeling. After more exploration, we can configure the kernel to use which pipeline in KernelConfig.cpp.

Reply:
Ok, let me move the state of this PR to draft to run some experiments before creating the new pipeline.